### PR TITLE
Tin-44-crud-address

### PR DESCRIPTION
**Pull request by Geofery**
Finished Crud Adress

### CODE REVIEW

File: src/DTO/addressDTO.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export interface AddressResponseDTO{
+    addressId: number;
+    city: string; 
+    zipcode: string;
+    address: string;
+}
+
+export interface AddressRequestDTO{
+    addressId: number;
```
Comment (Fladbrod) [inline src/DTO/addressDTO.ts:9]: Consider making addressId optional, that way you don't need AddressRequestDTO2
```suggestion
    addressId?: number;
```

File: src/DTO/addressDTO.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    city: string; 
+    zipcode: string;
+    address: string;
+}
+
+export interface AddressRequest2DTO{
+    city: string; 
+    zipcode: string;
+    address: string;
+}
```
Comment (Fladbrod) [inline src/DTO/addressDTO.ts:19]: Could be deleted
```suggestion
```

File: src/Services/addressService.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+export const getAddresses = async () => {
+    const addresses = await addressRepo.find();
+    const addressDTOs: AddressResponseDTO[] = addresses.map(address => convertToDTO(address));
+    return addressDTOs;
+};
+
+export const getAddressById = async (id: number) => {
+    const response = await addressRepo.findOneById(id);
```
Comment (Fladbrod) [inline src/Services/addressService.ts:21]: You are using a deprecated version of findOneById, change the name to use your method from addressRepo
```suggestion
    const response = await addressRepo.findOneByID(id);
```

File: src/Services/addressService.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+export const getAddresses = async () => {
+    try {
+        const addresses = await addressRepo.find();
+        const addressDTOs: AddressResponseDTO[] = addresses.map(address => convertToDTO(address));
+        return addressDTOs;
+
+    } catch (error) {
+        return error.message === 'Couldent find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```
Comment (Fladbrod) [inline src/Services/addressService.ts:22]: Typo
```suggestion
        return error.message === 'Couldn\'t find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```

File: src/Services/addressService.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+export const getAddressById = async (id: number) => {
+    try {
+        const response = await addressRepo.findOneByID(id);
+        if (!response) {
+            return { err: 'Invalid id' };
+        }
+        return convertToDTO(response);
+
+    } catch (error) {
+        return error.message === 'Couldent find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```
Comment (Fladbrod) [inline src/Services/addressService.ts:35]: Typo
```suggestion
        return error.message === 'Couldn\'t find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```

File: src/Services/addressService.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+export const updateAddress = async (addressDTO: AddressRequestDTO) => {
+    try {
+        if (!addressDTO) {
+            return { err: 'Invalid address DTO!' };
+        }
+        const response = await addressRepo.save(addressDTO);
+        return convertToDTO(response);
+
+    } catch (error) {
+        return error.message === 'Couldent find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```
Comment (Fladbrod) [inline src/Services/addressService.ts:48]: Typo
```suggestion
        return error.message === 'Couldn\'t find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
```

**Review by Fladbrod — COMMENTED** [review]
Enkelte typos, og tvivl om vi skal bruge try catch på simple get metoder. Måske har @Rasmus-Kibshede en mening

File: src/Services/addressService.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        const addresses = await addressRepo.find();
+        const addressDTOs: AddressResponseDTO[] = addresses.map(address => convertToDTO(address));
+        return addressDTOs;
+
+    } catch (error) {
+        return error.message === 'Couldent find any addresses!' ? { err: error.message } : { err: 'Something went wrong!- we are working on it!' };
+    }
+};
+
+export const getAddressById = async (id: number) => {
```
[the remark this answers is not in the document]
Comment (Rasmus-Kibshede) [reply src/Services/addressService.ts:26]: Der skal try catch alle steder der kan blive sendt en error/exception 🙂

**Review by Fladbrod — APPROVED** [review]

**Pull request merged**